### DISCUSSION TITLE
Appwrite easy custom installation tool

### DISCUSSION THREAD

**Discussion by mbos2** [appwrite/appwrite#2442] [Ideas]
My message copy-pasted from discord:
> Hey guys. 
> Did you maybe think of customizing appwrite installation? 
> Like, when we install appwrite, it install all those containers we got there ... but what if we don't want Tasks and Functions? 
> Or just Functions, but want to keep tasks? 
>
> Could we also have functions and tasks without database and users? 
>
> If it is possible, for us that aren't so familiar with docker and what docker containers are necessary for what service, could there be 
> some form where we can select services we want, and it would generate installation script that would install only services we want?

This could be:
- some form on the website under https://appwrite.io/docs/installation
- CLI tool
- Desktop tool
- Maybe it could also be updated directly from appwrite console? So we use the console UI to add/remove appwrite services, which should add/remove those service containers? (If this is even possible with docker apps, would be super cool!)

I imagine custom installation could be updated with another services, or removing services, either with the same tool or new installation script that will update the project but won't mess up the current data should nothing be removes. Hope this sentence makes sense :)

This idea was inspired by having a look at https://github.com/appwrite/lite

**Comment by eldadfux** [comment]
I agree this could be useful especially with all the planned adapters for the database, and storage services. Could be great to see this as a community initiative as well.

**Comment by Swinkid** [comment]
Something like https://start.spring.io/ but for Appwrite would be cool. Spits out a custom compse file or something I would've figured?
